src/main/java/org/greencodeinitiative/creedengo/java/checks/UseFilterBeforeSort.java

```diff
@@ -0,0 +1,77 @@
+package org.greencodeinitiative.creedengo.java.checks;
+
+import org.sonar.check.Rule;
+import org.sonar.plugins.java.api.IssuableSubscriptionVisitor;
+import org.sonar.plugins.java.api.tree.*;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+@Rule(key = "GCI91")
+public class UseFilterBeforeSort extends IssuableSubscriptionVisitor {
+    @Override
+    public List<Tree.Kind> nodesToVisit() {
+        return Collections.singletonList(Tree.Kind.METHOD_INVOCATION);
+    }
+
+    @Override
+    public void visitNode(Tree tree) {
+        if (tree instanceof MethodInvocationTree) {
+            MethodInvocationTree mit = (MethodInvocationTree) tree;
+
+            if (!isTerminalOperation(mit)) {
+                return; // Ignoring intermediate steps (filter, sorted, etc.)
+            }
+
+            List<String> methodChain = extractChainedMethodNames(mit);
+
+            int sortedIndex = methodChain.indexOf("sorted");
+            int filterIndex = methodChain.indexOf("filter");
+
+            if (sortedIndex != -1 && filterIndex != -1 && sortedIndex < filterIndex) {
```

Comment:
are you sure about your algorithms ... I think in streams we can have several select method ... thus you can control only the first one ? or more complex, control all the stream and we want only control th second part ... for example : 
stream
   . filtrer()
   .sorted()
   . findAny() /// first part
   . sort()
   . filter()
   . collect() // second part

what is the behaviour for examples with the same complexity (several parts) ?
can you add some such cases in your test resources files to control the behaviour ?